src/lib.rs

```diff
@@ -40,24 +48,69 @@ pub struct FastraceService<S> {
     service: S,
 }
 
-impl<S, Body> Service<Request<Body>> for FastraceService<S>
-where S: Service<Request<Body>>
+use axum::extract::Request;
+
+impl<S> Service<Request> for FastraceService<S>
+where
+    S: Service<Request, Response = Response> + Send + 'static,
+    S::Future: Send + 'static,
 {
     type Response = S::Response;
     type Error = S::Error;
-    type Future = fastrace::future::InSpan<S::Future>;
+    type Future =
+        Pin<Box<dyn Future<Output = Result<Self::Response, Self::Error>> + Send + 'static>>;
```

Comment:
Is it possible to return a concrete future type instead of wrapping it in a `Pin<Box>`.

Reply:
I see `InSpan` uses pin-project, I'll use that as well!